### PR TITLE
[HOPSWORKS-815] Mail from dataset request addressing the wrong user

### PR DESCRIPTION
**Pull request by robzor92**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [x] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by robzor92** [comment]
Jenkins testhttps://github.com/logicalclocks/hops-testing/pull/234

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java [hunk 1]
```diff
@@ -239,18 +243,26 @@ public Response requestJoin(RequestDTO requestDTO, @Context HttpServletRequest r
...
-            + "After loging in to hopsworks go to : /project" + project.getId()
-            + " and go to members tab "
-            + "if you want to add this person as a member in your project. \n";
+            + "Project name: " + project.getName() + "\n";
+
+    if(requestDTO.getMessageContent() != null) {
+      msg += "Attached message: " + requestDTO.getMessageContent() + "\n";
+    }
+
+    msg += "After loging in to Hopsworks go to : /project" + project.getId()
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java]: @robzor92 typo in `loging`

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java [hunk 2]
```diff
@@ -239,18 +243,26 @@ public Response requestJoin(RequestDTO requestDTO, @Context HttpServletRequest r
...
             + user.getFname() + " " + user.getLname()
             + " wants to join a project you own. \n\n"
-            + "Project name: " + project.getName() + "\n"
-            + "Attached message: " + requestDTO.getMessageContent() + "\n"
-            + "After loging in to hopsworks go to : /project" + project.getId()
-            + " and go to members tab "
-            + "if you want to add this person as a member in your project. \n";
+            + "Project name: " + project.getName() + "\n";
+
+    if(requestDTO.getMessageContent() != null) {
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java]: @robzor92 should we check for empty message as well here? as in `Strings.isNullorEmpty()`

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java [hunk 3]
```diff
@@ -239,18 +243,26 @@ public Response requestJoin(RequestDTO requestDTO, @Context HttpServletRequest r
...
 
     Users from = user;
     Users to = userFacade.findByEmail(project.getOwner().getEmail());
     String message = "Hi " + to.getFname() + "<br>"
             + "I would like to join a project you own. <br>"
-            + "Project name: " + project.getName() + "<br>"
-            + requestDTO.getMessageContent();
+            + "Project name: " + project.getName() + "<br>";
+
+    if(requestDTO.getMessageContent() != null) {
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java]: @robzor92 should we check for empty message as well here? as in Strings.isNullorEmpty()

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java (hunk 1 above)
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java]: done

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java (hunk 3 above)
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java]: done

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java (hunk 2 above)
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java]: done

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java [hunk 4]
```diff
@@ -66,6 +66,7 @@
 import io.hops.hopsworks.common.util.Settings;
 import io.hops.hopsworks.jwt.annotation.JWTRequired;
 import io.swagger.annotations.Api;
+import org.elasticsearch.common.Strings;
```
[the remark this answers is not in the document]
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/RequestService.java]: 10 usages of the elasticsearch infact, anyway I fixed this one

**Pull request merged**